### PR TITLE
ceph-osd-prestart.sh: drop Upstart-specific code

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/15984

### CODE REVIEW

**Comment by smithfarm** [comment]
@ktdreyer @b-ranto Care to review?

**Comment by ktdreyer** [comment]
LGTM (untested)

**Comment by b-ranto** [comment]
I'm not sure we want to do this. Especially when this is considered in conjunction with http://tracker.ceph.com/issues/15990 where we constantly get the ceph-disk failures because we treat this situation as an error and not just a warning.

I'd suggest to do the same for 15990 and just warn (not error out) in case one of the devices is not up, yet.

File: src/ceph-osd-prestart.sh [hunk 1]
```diff
@@ -21,15 +21,14 @@ data="/var/lib/ceph/osd/${cluster:-ceph}-$id"
...
 
 if [ -L "$journal" -a ! -e "$journal" ]; then
+    echo "ceph-osd(${cluster:-ceph}-$id): journal not present, not starting yet." 1>&2
     udevadm settle --timeout=5 || :
     if [ -L "$journal" -a ! -e "$journal" ]; then
-        echo "ceph-osd($UPSTART_INSTANCE): journal not present, not starting yet." 1>&2
-        stop
-        exit 0
+        echo "ceph-osd(${cluster:-ceph}-$id): journal still not present, aborting." 1>&2
+        exit 1
```
Comment (smithfarm) [inline src/ceph-osd-prestart.sh]: @b-ranto So, change this line to "exit 0" and perhaps change the warning in the previous line to "journal device still not present after 5 seconds; giving up" ?

Comment (b-ranto) [inline src/ceph-osd-prestart.sh]: I think we should be good if we just dropped the 'stop' line. The messages at their current form look sane to me.

**Comment by smithfarm** [comment]
@b-ranto OK, I narrowed the scope of this PR. It no longer attempts to fix http://tracker.ceph.com/issues/15984

**Comment by smithfarm** [comment]
Changelog:
* rebased
* narrowed scope of modifications

**Comment by smithfarm** [comment]
Hey Jenkins, test this please

**Comment by smithfarm** [comment]
@b-ranto Do you think this is OK to merge?

**Comment by b-ranto** [comment]
Yup, seems pretty safe (merged it myself).

**Pull request merged**